Invalid config errors swallowed

I just ran into a weird situation.

Here is my scalafmt config:

```hocon
align.ifWhileOpenParen = false
align.openParenCallSite = false
align.openParenDefnSite = false
align.tokens.add = ["*>", "<-"]
continuationIndent.defnSite = 2
continuationIndent.extendSite = 2
danglingParentheses = true
docstrings = ScalaDoc
importSelectors = singleLine
includeCurlyBraceInSelectChains = false
maxColumn = 120
newlines.alwaysBeforeElseAfterCurlyIf = false
newlines.alwaysBeforeTopLevelStatements = true
optIn.selfAnnotationNewline = false
optIn.configStyleArguments = false
rewrite.rules = ["AsciiSortImports", "ExpandImportSelectors", "PreferCurlyFors", "RedundantBraces", "RedundantParens", 'SortImports", "SortModifiers"]
spaces.afterTripleEquals = true
spaces.inImportCurlyBraces = true
version=2.0.1
```

When I ran `sbt scalafmtSbt` (other tasks had similar output), I got the following:

```shell
λ sbt scalafmtSbt
[info] Loading settings for project global-plugins from sbt-updates.sbt,metals.sbt ...
[info] Loading global plugins from /home/nequi/.sbt/1.0/plugins
[info] Loading settings for project scala-integration-build from plugins.sbt ...
[info] Loading project definition from /home/nequi/dev/scala-integration/project
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] Loading settings for project scala-integration from build.sbt ...
[info] Set current project to Integration Test (in build file:/home/nequi/dev/scala-integration/)
[info] ()
[info] ()
[error] null: /home/nequi/dev/scala-integration/build.sbt
[error] (Compile / scalafmtSbt) null: /home/nequi/dev/scala-integration/build.sbt
[error] Total time: 3 s, completed 6-Sep-2019 11:47:52 AM
```

So I grabbed the `scalafmt` CLI via `coursier` and ran `./scalafmt`

The output was a whole lot of these (essentially once per file in the project):

```java
org.scalafmt.cli.FailedToFormat: /home/nequi/dev/scala-integration/#.scalafmt.conf#
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.scalafmt.dynamic.utils.ReflectUtils$ClassReflectOps$.invokeStatic$extension(ReflectUtils.scala:34)
	at org.scalafmt.dynamic.ScalafmtReflect.parseConfigFromString(ScalafmtReflect.scala:71)
	at org.scalafmt.dynamic.ScalafmtReflect.parseConfig(ScalafmtReflect.scala:66)
	at org.scalafmt.dynamic.ScalafmtDynamic.$anonfun$parseConfig$1(ScalafmtDynamic.scala:148)
	at scala.util.Try$.apply(Try.scala:213)
	at org.scalafmt.dynamic.ScalafmtDynamic.parseConfig(ScalafmtDynamic.scala:148)
	at org.scalafmt.dynamic.ScalafmtDynamic.$anonfun$resolveConfigWithScalafmt$3(ScalafmtDynamic.scala:140)
	at scala.util.Either.flatMap(Either.scala:341)
	at org.scalafmt.dynamic.ScalafmtDynamic.$anonfun$resolveConfigWithScalafmt$2(ScalafmtDynamic.scala:139)
	at scala.util.Either.flatMap(Either.scala:341)
	at org.scalafmt.dynamic.ScalafmtDynamic.resolveConfigWithScalafmt(ScalafmtDynamic.scala:136)
	at org.scalafmt.dynamic.ScalafmtDynamic.$anonfun$resolveConfig$4(ScalafmtDynamic.scala:122)
	at scala.util.Try$.apply(Try.scala:213)
	at org.scalafmt.dynamic.utils.ReentrantCache.getOrAddToCache(ReentrantCache.scala:41)
	at org.scalafmt.dynamic.ScalafmtDynamic.resolveConfig(ScalafmtDynamic.scala:121)
	at org.scalafmt.dynamic.ScalafmtDynamic.formatDetailed(ScalafmtDynamic.scala:105)
	at org.scalafmt.dynamic.ScalafmtDynamic.format(ScalafmtDynamic.scala:67)
	at org.scalafmt.cli.ScalafmtDynamicRunner$.handleFile(ScalafmtDynamicRunner.scala:96)
	at org.scalafmt.cli.ScalafmtDynamicRunner$.$anonfun$run$4(ScalafmtDynamicRunner.scala:55)
	at org.scalafmt.cli.ScalafmtDynamicRunner$.$anonfun$run$4$adapted(ScalafmtDynamicRunner.scala:49)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at org.scalafmt.cli.ScalafmtDynamicRunner$.$anonfun$run$3(ScalafmtDynamicRunner.scala:49)
	at scala.util.control.Breaks.breakable(Breaks.scala:42)
	at org.scalafmt.cli.ScalafmtDynamicRunner$.run(ScalafmtDynamicRunner.scala:49)
	at org.scalafmt.cli.Cli$.run(Cli.scala:96)
	at org.scalafmt.cli.Cli$.mainWithOptions(Cli.scala:69)
	at org.scalafmt.cli.Cli$.main(Cli.scala:58)
	at org.scalafmt.cli.Cli.main(Cli.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at coursier.bootstrap.launcher.a.a(Unknown Source)
	at coursier.bootstrap.launcher.Launcher.main(Unknown Source)
Caused by: org.scalafmt.Error$InvalidScalafmtConfiguration: Failed to read configuration: java.lang.IllegalArgumentException: Incompatible rewrites: SortImports and ExpandImportSelectors
	at org.scalafmt.config.RewriteSettings.<init>(RewriteSettings.scala:21)
	at org.scalafmt.config.RewriteSettings$$anon$1.$anonfun$read$1(RewriteSettings.scala:15)
	at metaconfig.Configured.map(Configured.scala:18)
	at org.scalafmt.config.RewriteSettings$$anon$1.read(RewriteSettings.scala:15)
	at metaconfig.internal.NoTyposDecoder$$anon$1.read(NoTyposDecoder.scala:20)
	at metaconfig.internal.ConfGet$.getOrElse(ConfGet.scala:25)
	at metaconfig.Conf.getSettingOrElse(Conf.scala:28)
	at org.scalafmt.config.ScalafmtConfig$$anon$1.read(ScalafmtConfig.scala:179)
	at metaconfig.internal.NoTyposDecoder$$anon$1.read(NoTyposDecoder.scala:20)
	at metaconfig.internal.NoTyposDecoder$$anon$1.read(NoTyposDecoder.scala:20)
	at org.scalafmt.config.ScalafmtConfig$$anonfun$configReader$1.applyOrElse(ScalafmtConfig.scala:341)
	at org.scalafmt.config.ScalafmtConfig$$anonfun$configReader$1.applyOrElse(ScalafmtConfig.scala:323)
	at metaconfig.ConfDecoder$$anon$2.read(ConfDecoder.scala:65)
	at metaconfig.ConfDecoder.$anonfun$read$1(ConfDecoder.scala:17)
	at metaconfig.Configured.andThen(Configured.scala:30)
	at metaconfig.ConfDecoder.read(ConfDecoder.scala:17)
	at metaconfig.ConfDecoder.read$(ConfDecoder.scala:16)
	at metaconfig.ConfDecoder$$anon$2.read(ConfDecoder.scala:61)
	at org.scalafmt.config.Config$.$anonfun$fromConf$1(Config.scala:70)
	at metaconfig.Configured.andThen(Configured.scala:30)
	at org.scalafmt.config.Config$.fromConf(Config.scala:55)
	at org.scalafmt.config.Config$.fromHoconString(Config.scala:40)
	at org.scalafmt.config.Config$.fromHoconString(Config.scala:33)
	at org.scalafmt.Scalafmt$.parseHoconConfig(Scalafmt.scala:94)
	at org.scalafmt.Scalafmt.parseHoconConfig(Scalafmt.scala)
	... 38 more
```

Apparently, `SortImports` and `ExpandImportSelectors` are incompatible with one another.
I am surprised that is the case.
But regardless, it seems that the sbt plugin just swallows these errors?

Once I remove `SortImports`, everything is happy.

--- a/plugin/src/main/scala/org/scalafmt/sbt/ScalafmtSbtReporter.scala
+++ b/plugin/src/main/scala/org/scalafmt/sbt/ScalafmtSbtReporter.scala
@@ -7,14 +7,21 @@ import org.scalafmt.interfaces.ScalafmtReporter
 import sbt.internal.util.MessageOnlyException
 import sbt.util.Logger
 
+import scala.util.control.NoStackTrace
+
 class ScalafmtSbtReporter(log: Logger, writer: PrintWriter)
     extends ScalafmtReporter {
   override def error(file: Path, message: String): Unit = {
     throw new MessageOnlyException(s"$message: $file")
   }
 
-  override def error(file: Path, e: Throwable): Unit =
-    error(file, e.getMessage)
+  override def error(file: Path, e: Throwable): Unit = {
+    if (e.getMessage != null) {
+      error(file, e.getMessage)
+    } else {
+      throw new FailedToFormat(file.toString, e)
+    }
+  }
 
   override def excluded(file: Path): Unit =
     log.debug(s"file excluded: $file")
@@ -23,4 +30,8 @@ class ScalafmtSbtReporter(log: Logger, writer: PrintWriter)
     log.debug(s"parsed config (v$scalafmtVersion): $config")
 
   override def downloadWriter(): PrintWriter = writer
+
+  private class FailedToFormat(filename: String, cause: Throwable)
+      extends Exception(filename, cause)
+      with NoStackTrace
 }

